unifiednlp-base/src/main/java/org/microg/tools/selfcheck/NlpStatusChecks.java

```diff
@@ -33,53 +34,102 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 
 import static android.content.Context.LOCATION_SERVICE;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.location.Address;
 import static android.location.LocationManager.NETWORK_PROVIDER;
+import android.os.IBinder;
 import static org.microg.nlp.api.Constants.LOCATION_EXTRA_BACKEND_COMPONENT;
 import static org.microg.tools.selfcheck.SelfCheckGroup.Result.Negative;
 import static org.microg.tools.selfcheck.SelfCheckGroup.Result.Positive;
 import static org.microg.tools.selfcheck.SelfCheckGroup.Result.Unknown;
 
-public class NlpStatusChecks implements SelfCheckGroup {
-    private Location mLastLocation;
+public class NlpStatusChecks extends Service implements SelfCheckGroup {
```

Comment:
Don't you think it's better to have the service part in a separate class and a callback to the checks?

Reply:
@thuryn: I think you shouldn't change self check at all but just add some text like: "Binding isn't needed fro intent driven API" and "All Android versions works if intent driven API is used".